Relationship tests September 2020 (#59)
ONSdigital/eq-survey-runner-benchmark: doc/performance-testing/0003-relationship-tests-sep-2020/summary.md

# Relationship Test September 2020

## Purpose
To test the effect of relationship performance (response time), when increasing household members progressively up to 30.

## Test conditions

> 

Reviewer: ```suggestion
```

GETs average: 100ms
POSTs average: 125ms
All requests average: 112ms

**New instance**

GETs average: 100ms
POSTs average: 121ms
All requests average: 111ms

This confirmed that the new instance and load injector were configured and performing the same as the daily benchmark setup.

## Test settings

| Setting | Value |
| --- | ---|
| Clients per instance | 64 |
| Clients hatch rate   | 64 |
| Wait time minimum | 1 |
| Wait time maximum | 2 |
| Requests JSON | test_relationships.json |
| Runtime | 20m |
| last commit | [783f102e6fbeab11fbc221e9800143e2340ee09c](https://github.com/ONSdigital/eq-questionnaire-runner/commit/783f102e6fbeab11fbc221e9800143e2340ee09c)

## Results

| Household Members | GET (ms) | POST (ms) | Average (ms) |
|-------------------|----------|-----------|--------------|
| 5                 | 96       | 105       | 100          |
| 10                | 82       | 104       | 92           |
| 20                | 91       | 115       | 103          |
| 30 (1st test)     | 118      | 162       | 140          |
| 30 (2nd test)     | 114      | 150       | 132          |

- All results were recorded at the 99th percentile.

> - ≈135k requests were generated by each test.

Reviewer 2: ```suggestion
- Each test generated ≈135k requests.
```

> - No errors or failures were recorded in any of the test.

Reviewer 2: ```suggestion
- No errors or failures were recorded in any of the tests.
```

## Observations

- Having 5/10 households members made no difference, and in fact returned slightly lower figures than the benchmark/calibrated results. I would put this down to natural deviation in testing and not worth further investigation.

> - Having 20 members in a household had a slightly higher POST time, but its marginal.

Reviewer 2: ```suggestion
- Having 20 members in a household had a slightly higher POST time, but it‘s marginal.
```
